File: nj2k/testData/newJ2k/postProcessing/RedunduntTypeCastAndProhibitedInline.java
import java.util.Iterator;
import java.util.List;

public class C {

    public static void consume1(C c) {

    }

    public static void consume2(C c) {

    }

    public static void foo(List<C> cList) {
        for (Iterator iter = cList.iterator(); iter.hasNext();) {
            C c = (C) iter.next();
            consume1(c);
            consume2(c);
        }
    }
}

File: nj2k/testData/newJ2k/postProcessing/RedunduntTypeCastAndProhibitedInline.kt
object C {
    fun consume1(c: C?) {}

    fun consume2(c: C?) {}

    fun foo(cList: List<C?>) {
        val iter: Iterator<*> = cList.iterator()
        while (iter.hasNext()) {
            val c = iter.next() as C?
            consume1(c)
            consume2(c)
        }
    }
}